### ISSUE TITLE
Make it more clear when Registrations are disabled

### ISSUE THREAD

**Issue by DraconicNEO** [LemmyNet/lemmy-ui#2070]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

On Instances where account creation is disabled I feel like they should not show the sign-up form with username, email, and password fields because it can be confusing to people as it looks like the Instance is open when it isn't.
![gpe8j1fe8rjb1](https://github.com/LemmyNet/lemmy-ui/assets/44508470/18f90462-eefe-4a6e-9c6b-75205ce100c8)

As an example the person (credit to u/TheTwelveYearOld on Reddit for this screenshot) is trying to sign up to an Instance with closed registrations but due to how the UI works it's not entirely clear and to others not trying to sign up that registrations there are closed and it looks like Registrations are open.

My proposal is to make it so that when registrations are closed it'll show a message, similar to the application form but without a field to put in data explaining that Account creation has been disabled on this Instance and that users cannot create an account here. 

I made a mockup of what something like that might look like in an image editor:
![Lemmy Registration Disabled example](https://github.com/LemmyNet/lemmy-ui/assets/44508470/e379d923-3a62-4821-a3c3-1bcd765ea4b8)

**Comment by fatflyingpigs** [comment]
I think this is very much needed. It's annoying when the first interaction a person has with the site is trying to register, possibly putting the entry into a password manager, then hitting "sign up" only to get an error that should have disabled the form in the first place.

It would be nice if we could put a custom message to explain why registration is disabled. Message could say something about a temporary close (bot spam, etc.), that the instance is invite only, or another reason.

**Comment by DraconicNEO** [comment]
> I think this is very much needed. It's annoying when the first interaction a person has with the site is trying to register, possibly putting the entry into a password manager, then hitting "sign up" only to get an error that should have disabled the form in the first place.
> 
> It would be nice if we could put a custom message to explain why registration is disabled. Message could say something about a temporary close (bot spam, etc.), that the instance is invite only, or another reason.

Yeah, it's definitely very important Feedback, I can imagine it's very frustrating to people to try and sign up to an instance only to be told afterwards registration is closed, it would be much better for them to just see the message upfront that it's closed and why its closed.

Also there aren't really Invite-only lemmy instances yet since lemmy doesn't really have an invitation system unfortunately, though it has been requested here LemmyNet/lemmy#1777 not sure why it wasn't included early on though.
